### ISSUE TITLE
input selectionStart/selectionEnd incorrect during type()

### ISSUE THREAD

**Issue by DJTB** [cypress-io/cypress#1234]
Firstly, thanks for Cypress, it's great!

- Operating System: Ubuntu 16.04
- Cypress Version: 1.4.1
- Browser Version: Chrome 63

### Is this a Feature or Bug?
Bug

### Current behavior:
selectionStart and selectionEnd do not match expected cursor during `.type()` when position modified by  `{left|rightArrow}`

### Desired behavior:
selectionStart and selectionEnd **should match cursor _during_** `.type()`

### How to reproduce:
html file with `<input type="text" id="#input" />`

#### Test code:

```js
describe('musses up cursor position', () => {
  it('fails', () => {
    cy
      .visit('') // localhost
      .get('#input')
      .then(($el) => {
        $el.on('input', ({ target: { value, selectionStart, selectionEnd } }) =>
          console.log('typing', value, selectionStart, selectionEnd) // incorrect
        );
        return $el;
      })
      .clear()
      .type('abcde{leftArrow}{leftArrow}f')
      // these are not really what I'm testing 
      // I'm relying on these values during type()
      .should('have.prop', 'selectionStart', 4)
      .and('have.prop', 'selectionEnd', 4)
  });
});
```

### Additional Info (images, stack traces, etc)
![image](https://user-images.githubusercontent.com/5353151/35481304-261e40fe-0475-11e8-834d-46565cb9141b.png)

We have an input listener that converts latin chars `romaji` into kana `ろまじ` during input, and relies on selectionStart/End in order to handle any mid text insertion, and then replaces the cursor to the correct position after conversion (the input value length can change `ryo` => `りょ` 3 => 2). The functionality works correctly when manually testing in Chrome, but not with Cypress.

I can work around it for my particular use-case by forcing an update with the (manually calculated) correct position but ideally the input should have the correct values during typing:
```js
it('converts characters inserted mid-text after cursor movement', () => {
  // already bound input listener that does all the conversion handling
  cy
    .get('#input')
    .type('wanakana')
    .should('have.value', 'わなかな') // passes
    .type('{leftArrow}{leftArrow}shi') // now 'わなshiかな'
     // usually `shi` would be converted instantly after 
     // the `i` input event that is fired in type()
     // but selectionRange was incorrectly 7,7 at that time
    .then(($el) => {
      $el.get(0).setSelectionRange(5, 5);
      return $el;
    })
    .trigger('input')
    .should('have.value', 'わなしかな'); // passes only due to .then().trigger()
});
```

**Comment by Bkucera** [comment]
I'm actively working on this, a fix is in the works and should be out in an upcoming patch release. This issue should be fixed alongside other `cy.type` fixes mentioned in #1241 epic

**Comment by jennifer-shehane** [comment]
This fix was released in [Cypress 3.0.3](https://on.cypress.io/changelog#3-0-3)
